cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/ProvisionRecommendation.java

```diff
@@ -0,0 +1,397 @@
+  // If the resource is partition, the name of the topic must be specified.
+  protected final String _topic;
+  // A typical broker id and its capacity (one cannot be specified without the other)
+  protected final int _typicalBrokerId;
+  protected final double _typicalBrokerCapacity;
+  // Specific resource, such as {@link Resource#DISK}
+  protected final Resource _resource;
+  // Excluded racks -- i.e. racks for which brokers should not be added to or removed from
+  protected final Set<String> _excludedRackIds;
+  // Total resource capacity required to add or remove
+  protected final double _totalCapacity;
+
+  public static class Builder {
+    // Required parameters
+    private final ProvisionStatus _status;
+    // Optional parameters - initialized to default values
+    private int _numBrokers = DEFAULT_OPTIONAL_INT;
+    private int _numRacks = DEFAULT_OPTIONAL_INT;
+    private int _numDisks = DEFAULT_OPTIONAL_INT;
+    private int _numPartitions = DEFAULT_OPTIONAL_INT;
+    private String _topic = null;
+    private int _typicalBrokerId = DEFAULT_OPTIONAL_INT;
+    private double _typicalBrokerCapacity = DEFAULT_OPTIONAL_DOUBLE;
+    private Resource _resource = null;
+    private Set<String> _excludedRackIds = null;
+    private double _totalCapacity = DEFAULT_OPTIONAL_DOUBLE;
+
+    public Builder(ProvisionStatus status) {
+      if (!(status == ProvisionStatus.UNDER_PROVISIONED || status == ProvisionStatus.OVER_PROVISIONED)) {
+        throw new IllegalArgumentException(String.format("Provision recommendation is irrelevant for provision status %s.", status));
+      }
+      _status = status;
+    }
+
+    /**
+     * (Optional) Set number of brokers
+     * @param numBrokers Number of brokers to add or remove
+     * @return this builder.
+     */
+    public Builder numBrokers(int numBrokers) {
+      if (numBrokers <= 0) {
+        throw new IllegalArgumentException(String.format("Number of brokers must be positive (%d).", numBrokers));
+      }
+      _numBrokers = numBrokers;
+      return this;
+    }
+
+    /**
+     * (Optional) Set number of racks
+     * @param numRacks Number of racks to add or remove
+     * @return this builder.
+     */
+    public Builder numRacks(int numRacks) {
+      if (numRacks <= 0) {
+        throw new IllegalArgumentException(String.format("Number of racks must be positive (%d).", numRacks));
+      }
+      _numRacks = numRacks;
+      return this;
+    }
+
+    /**
+     * (Optional) Set number of disks
+     * @param numDisks Number of disks to add or remove
+     * @return this builder.
+     */
+    public Builder numDisks(int numDisks) {
+      if (numDisks <= 0) {
+        throw new IllegalArgumentException(String.format("Number of disks must be positive (%d).", numDisks));
+      }
+      _numDisks = numDisks;
+      return this;
+    }
+
+    /**
+     * (Optional) Set number of partitions
+     * @param numPartitions Number of partitions to add or remove
```

Comment:
```suggestion
     * @param numPartitions The recommended number of partitions.
```
Can we align the definition of this field with [`admin.NewTopic`](https://kafka.apache.org/26/javadoc/org/apache/kafka/clients/admin/NewTopic.html#numPartitions--)?
>  The number of partitions for the new topic or -1 if a replica assignment has been specified.

### Reason
By aligning the two, users can pass this value directly into `KafkaCruiseControlUtils.wrapTopic(int partitionCount)` when updating the partition count. 
Users would save the step to query the current partition count to add this number to pass this parameter.

Reply:
Makes sense.